apis/batch/v1alpha1/task_types.go

```diff
@@ -15,12 +15,23 @@
 package v1alpha1
 
 import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 var BatchTaskGVK = GroupVersion.WithKind("BatchTask")
 
+type Parent struct {
```

Comment:
Nit: Isn't this `*parent.ProjectAndLocationRef`

Reply:
It is. But this PR should only contain code for  BatchJob promotion.